x-pack/legacy/plugins/maps/public/selectors/map_selectors.js

```diff
@@ -125,6 +125,21 @@ export const getRefreshConfig = ({ map }) => {
 
 export const getRefreshTimerLastTriggeredAt = ({ map }) => map.mapState.refreshTimerLastTriggeredAt;
 
+export function getLayerDescriptor(state = {}, layerId) {
+  const layerListRaw = getLayerListRaw(state);
+  return layerListRaw.find(layer => layer.id === layerId);
+}
+
+export function getDataRequest(state = {}, layerId, dataId) {
```

Comment:
rename to `getDataRequestDescriptor` to distinguish this from the actual type (https://github.com/elastic/kibana/blob/8e9a8a84dccfa7965ce8a22362885e6cdef8b51f/x-pack/legacy/plugins/maps/public/layers/util/data_request.js#L8) use elsewhere